test/ResearcherContract.test.js

```diff
@@ -0,0 +1,136 @@
+const ResearcherContract = artifacts.require("ResearcherContract");
+const UserContract = artifacts.require("UserContract");
+
+const expectRevert = require("@openzeppelin/test-helpers").expectRevert;
+
+contract("ResearcherContract", (accounts) => {
+  let instance;
+  let userContract;
+  let [ownerAddress, resea1Address, resea2Address, resea3Address] = accounts;
+
+  const addResearcher = async (name, address) => {
+    await instance.addResearcher(
+      name,
+      "111.111.111-00",
+      "CPF",
+      "Brazil",
+      "SP",
+      "Jundiai",
+      "135465-005",
+      {from: address}
+    );
+  };
+
+  beforeEach(async () => {
+    userContract = await UserContract.new();
+
+    instance = await ResearcherContract.new(userContract.address);
+
+    await userContract.newAllowedCaller(instance.address);
+    await instance.newAllowedUser(resea1Address);
+    await instance.newAllowedUser(resea2Address);
+
+});
+
+context("when will create new researcher (.addResearcher)", () => {
+    context("when is not an allowed user", () => {
+        it("should return error message", async () => {
+            await expectRevert(
+                addResearcher("Reseacher C", resea3Address),
+                "Not allowed user"
+            );
+        })
+    }
+    
+    );
+    context("when is and allowed user", () => {
+        context("when researcher exists", () => {
+        it("should return error", async () => {
```

Comment:
O it deve fica identado pra dentro do seu context com 2 espaços

Reply:
Corrigido.